### PR TITLE
feat(benchmark): separate benchmarks by number of cores

### PR DESCRIPTION
**Pull request by mehulkar**
- [x] Move ttft scripts to a subdir to group them together
- [x] Move `fold.jq` logic into `ttft/index.ts`
- [x] Separate the data by number of cores

Closes TURBO-1685

### CODE REVIEW

File: benchmark/src/ttft/index.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+if (!fs.existsSync(TURBO_BIN)) {
+  throw new Error("No turbo binary found");
+}
+
+cp.execSync(`${TURBO_BIN} --version`, { stdio: "inherit" });
+
+run(fullProfilePath); // Actual benchmark
+
+const profileJSON = JSON.parse(fs.readFileSync(fullProfilePath).toString());
```
Comment (chris-olszewski) [inline benchmark/src/ttft/index.ts:48]: We need to define the schema of this JSON as a Typscript interface and that should clear up all of the TS errors.
```suggestion
const profileJSON = JSON.parse(fs.readFileSync(fullProfilePath).toString()) as ProfileJson;
```

**Review by chris-olszewski — CHANGES_REQUESTED** [review]
Other than that TS errors I think this looks good.

File: benchmark/src/ttft/tinybird.ts [hunk 2]
```diff
@@ -16,8 +16,19 @@ if (!runID) {
...
 const DATA_SOURCE_URL = `https://api.us-east.tinybird.co/v0/events?name=${DATA_SOURCE_NAME}`;
 
+type TinyBirdTTFT = Omit<TTFTData, "cpus">;
+
 async function main() {
   const data = getTTFTData(filePath, runID);
+
+  const augmentData: TinyBirdTTFT = {
+    ...data,
+    platform: `${data.platform}-${data.cpus}-cores`,
```
[the remark this answers is not in the document]
Comment (mehulkar) [reply benchmark/src/ttft/tinybird.ts:26]: They are, this TB script happens after `<profile>-ttft.json` is generated. We need different data for Slack and TB

File: benchmark/src/ttft/index.ts (hunk 1 above)
Comment (mehulkar) [reply benchmark/src/ttft/index.ts:48]: i could do that, didn't seem super useful tbh and each item in profile.json can look different i think, especially `args`

File: benchmark/src/ttft/tinybird.ts [hunk 3]
```diff
@@ -16,8 +16,19 @@ if (!runID) {
...
+  };
+
+  // @ts-expect-error don't know how to fix this
+  delete augmentData.cpus;
```
[the remark this answers is not in the document]
Comment (mehulkar) [reply benchmark/src/ttft/tinybird.ts]: that works, updated

File: benchmark/src/ttft/tinybird.ts [hunk 4]
```diff
@@ -16,8 +16,19 @@ if (!runID) {
 const DATA_SOURCE_NAME = "turborepo_perf_ttft";
 const DATA_SOURCE_URL = `https://api.us-east.tinybird.co/v0/events?name=${DATA_SOURCE_NAME}`;
 
+type TinyBirdTTFT = Omit<TTFTData, "cpus">;
+
 async function main() {
   const data = getTTFTData(filePath, runID);
+
+  const augmentData: TinyBirdTTFT = {
```
[the remark this answers is not in the document]
Comment (mehulkar) [reply benchmark/src/ttft/tinybird.ts:26]: missed that I guess, updated

File: benchmark/src/ttft/tinybird.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mehulkar) [reply benchmark/src/ttft/tinybird.ts]: running again with some fixes: https://github.com/vercel/turbo/actions/runs/6896943986

I definitely saw `*-n-core` platform values in TB before, but maybe that was from local. But yes `-ttft.json` is before the munging for TB

**Review by chris-olszewski — APPROVED** [review]

File: benchmark/src/ttft/tinybird.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mehulkar) [reply benchmark/src/ttft/tinybird.ts]: looks good now: https://github.com/vercel/turbo/actions/runs/6897363423/job/18765919652#step:7:12

**Pull request merged**